systemcontractindex/stakingindex/event_handler.go

```diff
@@ -36,61 +31,21 @@ type eventHandler struct {
 	dirty           *cache             // dirty cache, a view for current block
 	delta           batch.KVStoreBatch // delta for db to store buckets of current block
 	tokenOwner      map[uint64]address.Address
+	// context for event handler
+	block *block.Block
```

Comment:
seems only `block.Height()` is ever used in all handlers, so can change to `blkHeight uint64`